code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/OpenLDAP/OpenLDAPConfigManager.java

```diff
@@ -36,105 +44,114 @@ protected Object createInternal(String type, ApiRequest request) {
         }
         LDAPConstants config = request.proxyRequestObject(LDAPConstants.class);
         LDAPUtils.validateConfig(config);
-        return updateCurrentConfig(config);
+        Map<String, Object> configMap = CollectionUtils.toMap(request.getRequestObject());
+        return updateCurrentConfig(configMap);
```

Comment:
Originally we were using reflection to translate the request -> LDAPConstants object. That translation does not work after adding a complex field List<Identity> allowedIdentities, which is sent in the API request as a List<Map<String, String>>. So now we just read directly from the API request Map